src/main/java/chess/domain/board/ChessBoard.java

```diff
@@ -0,0 +1,83 @@
+package chess.domain.board;
+
+import chess.domain.piece.Bishop;
+import chess.domain.piece.Color;
+import chess.domain.piece.King;
+import chess.domain.piece.Knight;
+import chess.domain.piece.Pawn;
+import chess.domain.piece.Piece;
+import chess.domain.piece.Queen;
+import chess.domain.piece.Rook;
+import chess.domain.position.File;
+import chess.domain.position.Position;
+import chess.domain.position.Rank;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public class ChessBoard {
+    private final Map<Position, Piece> chessBoard = new LinkedHashMap<>();
+
+    public ChessBoard() {
+        initializeBlackPieces();
+        initializeWhitePieces();
+    }
+
+    public void move(Position source, Position target) {
+        if (!canMove(source, target)) {
+            throw new IllegalArgumentException("올바르지 않은 이동입니다.");
+        }
+        Piece sourcePiece = chessBoard.get(source);
+        chessBoard.put(target, sourcePiece);
+        chessBoard.remove(source);
+    }
+
+    public Map<Position, Piece> getChessBoard() {
+        return chessBoard;
+    }
+
+    private boolean canMove(Position source, Position target) {
+        if (!chessBoard.containsKey(source)) {
+            throw new IllegalArgumentException("이동할 수 있는 말이 없습니다.");
+        }
+
+        Piece sourcePiece = chessBoard.get(source);
+        Color targetPieceColor = Color.NONE;
+
+        if (chessBoard.containsKey(target)) {
+            Piece targetPiece = chessBoard.get(target);
+            targetPieceColor = targetPiece.getColor();
+        }
```

Comment:
```suggestion
        Piece targetPiece = chessBoard.get(target);
        Color targetPieceColor = Optional.ofNullable(targetPiece)
            .map(Piece::getColor).orElse(Color.NONE);
```
할당된 변수를 변경하기보다는 생성할때 결정하도록 짜면 더 좋을것 같습니다!

Reply:
Optional을 사용해서 한번에 처리할수도 있네요! 반영하였습니다. 감사합니다.